### PR TITLE
kernel: bump 5.15 to 5.15.76

### PR DESCRIPTION
**Pull request by graysky2**
All patches automatically rebased.  I emailed the Jonathan Bell to check my manual rebase for accuracy.

Build system: x86_64
Build-tested: bcm2711/RPi4B, mt7622/RT3200
Run-tested: bcm2711/RPi4B, mt7622/RT3200

Signed-off-by: John Audia <therealgraysky@proton.me>

### CODE REVIEW

**Comment by Ansuel** [comment]
@graysky2 new flag :D

File: target/linux/bcm27xx/patches-5.15/950-0600-xhci-quirks-add-link-TRB-quirk-for-VL805.patch [hunk 1]
```diff
@@ -48,13 +48,3 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
  
  	/* Don't update the ring cycle state for the producer (us). */
  	addr = xhci_trb_virt_to_dma(new_seg, new_deq);
---- a/drivers/usb/host/xhci.h
-+++ b/drivers/usb/host/xhci.h
-@@ -1903,6 +1903,7 @@ struct xhci_hcd {
- #define XHCI_NO_SOFT_RETRY	BIT_ULL(40)
- #define XHCI_BROKEN_D3COLD	BIT_ULL(41)
- #define XHCI_EP_CTX_BROKEN_DCS	BIT_ULL(42)
-+#define XHCI_AVOID_DQ_ON_LINK	BIT_ULL(43)
```
Comment (hauke) [inline target/linux/bcm27xx/patches-5.15/950-0600-xhci-quirks-add-link-TRB-quirk-for-VL805.patch:57]: This is wrong `XHCI_AVOID_DQ_ON_LINK` was not added in v5.15.75, but XHCI_SUSPEND_RESUME_CLKS was added. See https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?id=f002aa7c0ac5518a152ec24503c9f887031984a0 This change is still needed, but you have to it a new number , increase it by 1. There are 2 other files with this problem.

Comment (graysky2) [reply target/linux/bcm27xx/patches-5.15/950-0600-xhci-quirks-add-link-TRB-quirk-for-VL805.patch:57]: Yes!  This is what I did not know.  So keep the numbers from upstream, then add the next largest number for this. I will do that and rebase.

Comment (graysky2) [reply target/linux/bcm27xx/patches-5.15/950-0600-xhci-quirks-add-link-TRB-quirk-for-VL805.patch:57]: @hauke @Ansuel - Please review my latest push wherein I believe I corrected this mistake.

**Comment by graysky2** [comment]
I noticed a a build error on mediatek/filogic and rockchip/armv8.  I added the change to `
target/linux/generic/config-5.15` as you see. Is this the right action, to disable this ERRATUM symbol?

File: target/linux/bcm27xx/patches-5.15/950-0600-xhci-quirks-add-link-TRB-quirk-for-VL805.patch (hunk 1 above)
Comment (graysky2) [reply target/linux/bcm27xx/patches-5.15/950-0600-xhci-quirks-add-link-TRB-quirk-for-VL805.patch:57]: I am going to mark this as resolved.  Boot on my RPi4 with no dmesg regressions/USB seems to operate as expected.

**Comment by robimarko** [comment]
By default, it should be disabled as it only affects Cortex A55 cores, and I dont think we have any A55 SoC-s so far.
Filogic is A53 while Rockchip should be A72+A53 so none of them are affected

**Comment by graysky2** [comment]
Thanks for the confirmation, @robimarko

**Comment by graysky2** [comment]
Does anyone with CI experience know if it is possible to not rebuilt the 70 targets when the author just updates the commit message?  Seems wasteful.  For example, I just edited the message to reflect my run testing on another device and it seems like the entire set of checks are rebuilding their respective images.

**Comment by Ansuel** [comment]
@graysky2 a dev can cancel the old build but why bother honestly...

**Comment by graysky2** [comment]
You are probably right.  Just seems like wasted electrons to me is all.

**Comment by Borromini** [comment]
Compile-tested: mvebu/cortexa72
Run-tested: mvebu/cortexa72 (RB5009UG+S+IN)

File: target/linux/bcm27xx/patches-5.15/950-0600-xhci-quirks-add-link-TRB-quirk-for-VL805.patch (hunk 1 above)
Comment (hauke) [reply target/linux/bcm27xx/patches-5.15/950-0600-xhci-quirks-add-link-TRB-quirk-for-VL805.patch:57]: looks good now

**Comment by graysky2** [comment]
@robimarko - 5.15.75 to 5.15.76 introduces a [new symbol](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/diff/arch/arm64/Kconfig?id=v5.15.76&id2=v5.15.75) which applies to both Cortex-A57 and Cortex-A72.  I pushed this initially disabling the symbol, but RPi4 is Cortex-A72 I believe.  Shouldn't we actually enable it there?  If so, how many other targets need it to be enabled?

**Comment by robimarko** [comment]
According to the description RPi is not affected as it doesn't have crypto extensions, but there are A72 targets like mvebu that are affected and for those it should be enabled.

**Comment by graysky2** [comment]
@robimarko - Please review my latest push.  Is adding the symbol to `target/linux/mvebu/cortexa72/config-5.15` the correct move?

**Comment by robimarko** [comment]
Yes, that is correct, however, there should be more A72 targets in OpenWrt.

**Comment by graysky2** [comment]
@robimarko - I am unsure how to identify them and then know if they have supported crypto extensions (I actually thought RPi4 did).

**Comment by robimarko** [comment]
The only ones that are A72 in OpenWrt that I know of are Layerscape ARMv8 as some models are A72 and Rockchip as RK3399 is A72+A53 SoC.

Maybe somebody else knows if I have missed some, if its unclear if crypto extensions are supported enabling this should not do anything, but RPi for sure doesn't have crypto extensions as that is an optional extension that you gotta pay a license for and it will always be advertised in the docs

**Comment by graysky2** [comment]
If that is the case, layerscape/armv8_64b is currently on the 5.10 series of kernel so out-of-scope.  I will add to rockchip/armv8 and push an update.

**Comment by Ninjanoir78** [comment]
seems not working with R7800 ipq806x ??

patch

**Comment by trippleflux** [comment]
@graysky2 
Any chance for including x86_64 platform?, was trying to built 5.15.75 for x86, there is no error but just wanted to make sure if x86 patches is UpToDate.

**Comment by graysky2** [comment]
@trippleflux - all supported targets should be updated.  If your x86_64 build completed without error, you are good.

**Comment by Borromini** [comment]
For 5.51.76:

Compile-tested: mvebu/cortexa72
Run-tested: mvebu/cortexa72 (RB5009UG+S+IN)

**Comment by hauke** [comment]
Thank you for the patches, I applied them to master in a133423c594fb5d1a7f71c9474d4f550f426f800, a34255b795bcd939a66f11f100e0de8f54fd3707, 8e7cc062541397269095799f4111abb7eb3e90f7, 86ba286766c9ba6de34d5f4dab234dcc7010000c, af1e3439a8006a4878322f8129a51b23fa8be2b3 and 910bdda6afa229e35ef6e1f74795ee79aa99c2e2.

**Pull request closed without merging**